components/mediation/data-publishers/org.wso2.micro.integrator.prometheus.publisher/src/main/java/org/wso2/micro/integrator/prometheus/publisher/service/MetricResource.java

```diff
@@ -85,4 +99,18 @@ public boolean invoke(MessageContext synCtx) {
         return true;
     }
 
+    private Set<String> parseQuery(String query) throws IOException {
+
+        Set<String> names = new HashSet<>();
+        if (query != null) {
+            String[] pairs = query.split("&");
+            for (String pair : pairs) {
+                int idx = pair.indexOf("=");
+                if (idx != -1 && URLDecoder.decode(pair.substring(0, idx), "UTF-8").equals("name[]")) {
```

Comment:
Is there a possibility of null return? If so, shall we make this "name[]'.equals(URLDecoder.decode(pair.substring(0, idx), "UTF-8"))"?

Reply:
Can't we do what we have done in org.wso2.micro.integrator.management.apis.Utils.getQueryParameter(...)?

Reply:
Prometheus allows us to add "?name[]=metric_name_to_filter_by" to the end of the /metrics URL, in case we don't want the client library to return all metric names.